source/validate_id.cpp

```diff
@@ -740,7 +740,7 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
            return false);
   auto pointerIndex = 3;
   auto pointer = usedefs_.FindDef(inst->words[pointerIndex]);
-  if (!pointer.first || !spvOpcodeIsPointer(pointer.second.opcode)) {
+  if (!pointer.first || !spvOpcodeIsPotentialPointer(pointer.second.opcode)) {
```

Comment:
This isn't quite right.  The check here should be whether the _type_ of the `pointer` ID is an `OpTypePointer` result.  That cannot be determined from the opcode alone.  Please consider something like this instead:

```
auto pointerType = usedefs_.FindDef(pointer.second.type_id);
if (!pointerType.first || pointerType.second.opcode != SpvOpTypePointer) {
...
```